tests/test_operations.py

```diff
@@ -0,0 +1,342 @@
+    symlink_operation: SymlinkOperation, mock_src: str, mock_dest: str
+):
+    """Validate SymlinkOperation init"""
+    assert isinstance(symlink_operation, SymlinkOperation)
+    assert symlink_operation.src == mock_src
+    assert symlink_operation.dest == mock_dest
+
+
+def test_symlink_operation_format(
+    symlink_operation: SymlinkOperation,
+    mock_src: str,
+    mock_dest: str,
+    generation_context: GenerationContext,
+):
+    """Validate SymlinkOperation.format"""
+    exec = symlink_operation.format(generation_context)
+    assert isinstance(exec, Command)
+    assert str(mock_src) in exec.command
+    assert symlink_cmd in exec.command
+
+    normalized_path = os.path.normpath(mock_src)
+    parent_dir = os.path.basename(normalized_path)
+    final_dest = create_final_dest(generation_context.job_root_path, mock_dest)
+    new_dest = os.path.join(final_dest, parent_dir)
+    assert new_dest in exec.command
+
+
+def test_init_configure_operation(
+    configure_operation: ConfigureOperation, mock_src: str, mock_dest: str
+):
+    """Validate ConfigureOperation init"""
+    assert isinstance(configure_operation, ConfigureOperation)
+    assert configure_operation.src == mock_src
+    assert configure_operation.dest == mock_dest
+    assert configure_operation.tag == ";"
+    decoded_dict = base64.b64decode(configure_operation.file_parameters.encode("ascii"))
+    unpickled_dict = pickle.loads(decoded_dict)
+    assert unpickled_dict == {"FOO": "BAR"}
+
+
+def test_configure_operation_format(
+    configure_operation: ConfigureOperation,
+    mock_src: str,
+    mock_dest: str,
+    generation_context: GenerationContext,
+):
+    """Validate ConfigureOperation.format"""
+    exec = configure_operation.format(generation_context)
+    assert isinstance(exec, Command)
+    assert str(mock_src) in exec.command
+    assert configure_cmd in exec.command
+    assert create_final_dest(mock_src, mock_dest) in exec.command
+
+
+def test_init_file_sys_operation_set(file_system_operation_set: FileSysOperationSet):
+    """Test initialize FileSystemOperationSet"""
+    assert isinstance(file_system_operation_set.operations, list)
+    assert len(file_system_operation_set.operations) == 3
+
+
+def test_add_copy_operation(
+    file_system_operation_set: FileSysOperationSet, copy_operation: CopyOperation
+):
+    """Test FileSystemOperationSet.add_copy"""
+    assert len(file_system_operation_set.copy_operations) == 1
+    file_system_operation_set.add_copy(src=pathlib.Path("src"))
+    assert len(file_system_operation_set.copy_operations) == 2
+
+
+def test_add_symlink_operation(
+    file_system_operation_set: FileSysOperationSet, symlink_operation: SymlinkOperation
+):
+    """Test FileSystemOperationSet.add_symlink"""
+    assert len(file_system_operation_set.symlink_operations) == 1
+    file_system_operation_set.add_symlink(src=pathlib.Path("src"))
+    assert len(file_system_operation_set.symlink_operations) == 2
```

Comment:
I'll leave to your own discretion, but it might be a good idea to nut check the number of operations at the begining in case someone updates the fixture down the line. That way this test will continue to pass
```suggestion
    """Test FileSystemOperationSet.add_symlink"""
   orig_num_ops =  len(file_system_operation_set.symlink_operations)
    file_system_operation_set.add_symlink(src=pathlib.Path("src"))
    assert len(file_system_operation_set.symlink_operations) == orig_num_ops + 1
```

Reply:
A similar comment applies for copy and configure.

Reply:
thats a much better idea!